atest/testdata/standard_libraries/string/generate_random_string.robot

```diff
@@ -10,11 +10,29 @@ ${NUMBERS}        1234567890
 *** Test Cases ***
 Generate Random String With Defaults
     ${result} =    Generate Random String
-    String Lenght Should Be And It Should Consist Of    ${result}    8    ${LETTERS}${NUMBERS}
+    String Length Should Be And It Should Consist Of    ${result}    8    ${LETTERS}${NUMBERS}
 
 Generate Random String With Empty Length
     ${result} =    Generate Random String    ${EMPTY}    abc
-    String Lenght Should Be And It Should Consist Of    ${result}    8    abc
+    String Length Should Be And It Should Consist Of    ${result}    8    abc
+
+Generate Random String With Random Length
+    FOR    ${i}    IN RANGE    0    20
+        ${result} =    Generate Random String    5-10
+        String Length Should Be Within    ${result}    5    10
+    END
+
+Generate Random String With Invalid Ranges
+    # Cannot convert 'length' to integer
+    Run Keyword And Expect Error    ValueError: *    Generate Random String    5-
+    Run Keyword And Expect Error    ValueError: *    Generate Random String    foo-10
+    Run Keyword And Expect Error    ValueError: *    Generate Random String    5-bar
+    Run Keyword And Expect Error    ValueError: *    Generate Random String    foo-bar
+    Run Keyword And Expect Error    ValueError: *    Generate Random String    -
+    # Length is not a valid range of integers
+    ${range error} =    Set Variable    ValueError: Length is not a valid range of integers: *
+    Run Keyword And Expect Error    ${range error}    Generate Random String    --
+    Run Keyword And Expect Error    ${range error}    Generate Random String    1-2-3
```

Comment:
This could possibly be handled using a FOR loop. Something like this:
```
FOR    ${range}    IN    5-    foo-10    # ...
    Run Keyword And Expect Error
    ...    ValueError: Cannot convert 'length' argument '${range}' to an integer.
    ...    Generate Random String    ${range}
END
```
The above assumes checking the range has been changed as I propose in another comment and invalid ranges are validated in `_convert_to_integer`.